src/tacf/tacf.hpp

```diff
@@ -371,7 +372,15 @@ class Tacf : TargetedAcf
             return tacfFail;
         }
 
-        return writeFile(acf, size, acfFilePath);
+        int rc = writeFile(acf, size, acfFilePath);
+
+        // Enable the service user account using dbus interface.
+        TacfDbus().enableUser(serviceName);
+
+        // Unlock service user account using dbus interface.
+        TacfDbus().unlockUser(serviceName);
```

Comment:
Always unlock? Unlock required only it's locked?

Reply:
Yes always unlock.  The code could check if the user account is locked out because of too many bad password attempts, and then only unlock if it was locked.  But that would be extra logic and another D-Bus call.
Notice that the resetAdmin() function above does not check first; it just unlocks the password lockout.

Reply:
Ok